centreon/lib/Centreon/Object/Host/Host.php

```diff
@@ -45,4 +46,245 @@ class Centreon_Object_Host extends Centreon_Object
+        );
+
+        return $readVaultConfigurationRepository->findDefaultVaultConfiguration();
+    }
+
+    /**
+     * Get Client Token for Vault.
+     *
+     * @param VaultConfiguration $vaultConfiguration
+     * @param CentreonUserLog $centreonLog
+     * @param CentreonRestHttp $httpClient
+     * @return string
+     * @throws \Exception
+     */
+    private function authenticateToVault(
+        VaultConfiguration $vaultConfiguration,
+        CentreonUserLog $centreonLog,
+        CentreonRestHttp $httpClient
+    ): string {
+        try {
+            $url = $vaultConfiguration->getAddress() . ':' . $vaultConfiguration->getPort() . '/v1/auth/approle/login';
+            $body = [
+                "role_id" => $vaultConfiguration->getRoleId(),
+                "secret_id" => $vaultConfiguration->getSecretId(),
+            ];
+            $centreonLog->insertLog(5, 'Authenticating to Vault: ' . $url);
+            $loginResponse = $httpClient->call($url, "POST", $body);
+        } catch (\Exception $ex) {
+            $centreonLog->insertLog(5, $url . " did not respond with a 200 status");
+            throw $ex;
+        }
+
+        if (! isset($loginResponse['auth']['client_token'])) {
+            $centreonLog->insertLog(5, $url . " Unable to retrieve client token from Vault");
+            throw new \Exception('Unable to authenticate to Vault');
+        }
+
+        return $loginResponse['auth']['client_token'];
+    }
+
+    /**
+     * Get host secrets data from vault
+     *
+     * @param VaultConfiguration $vaultConfiguration
+     * @param integer $hostId
+     * @param string $clientToken
+     * @param CentreonUserLog $centreonLog
+     * @param CentreonRestHttp $httpClient
+     * @return array<string, mixed>
+     * @throws \Throwable
+     */
+    private function getHostSecretsFromVault(
+        VaultConfiguration $vaultConfiguration,
+        int $hostId,
+        string $clientToken,
+        CentreonUserLog $centreonLog,
+        CentreonRestHttp $httpClient
+    ): array {
+        $url = $vaultConfiguration->getAddress() . ':' . $vaultConfiguration->getPort() . '/v1/'
+            . $vaultConfiguration->getStorage() . '/centreon/hosts/' . $hostId;
+        $centreonLog->insertLog(5, sprintf("Search Host %d secrets at: %s", $hostId, $url));
+        $hostSecrets = [];
+        try {
+            $content = $httpClient->call($url, 'GET', null, ['X-Vault-Token: ' . $clientToken]);
+        } catch (\RestNotFoundException $ex) {
+            $centreonLog->insertLog(5, sprintf("Host %d not found in vault", $hostId));
+
+            return $hostSecrets;
+        } catch (\Exception $ex) {
+            $centreonLog->insertLog(5, sprintf('Unable to get secrets for host : %d', $hostId));
+            throw $ex;
+        }
+        if (array_key_exists('data', $content)) {
+            $hostSecrets = $content['data'];
+        }
+
+        return $hostSecrets;
```

Comment:
Seems less weird to me to explicit the retun values.
```suggestion
        try {
            $content = $httpClient->call($url, 'GET', null, ['X-Vault-Token: ' . $clientToken]);
        } catch (\RestNotFoundException $ex) {
            $centreonLog->insertLog(5, sprintf("Host %d not found in vault", $hostId));

            return [];
        } catch (\Exception $ex) {
            $centreonLog->insertLog(5, sprintf('Unable to get secrets for host : %d', $hostId));
            throw $ex;
        }
        if (array_key_exists('data', $content)) {
            return $content['data'];
        }

        return [];
```